multiversx_sdk/core/transactions_factories/transfer_transactions_factory.py

```diff
@@ -76,3 +79,48 @@ def create_transaction_for_esdt_token_transfer(self,
             gas_limit=extra_gas_for_transfer,
             add_data_movement_gas=True
         ).build()
+
+    def create_transaction_for_token_transfer(self,
+                                              sender: IAddress,
+                                              receiver: IAddress,
+                                              native_amount: Optional[int] = None,
+                                              token_transfers: Optional[Sequence[ITokenTransfer]] = None,
+                                              data: Optional[bytes] = None) -> Transaction:
+        if not native_amount and not token_transfers:
+            raise BadUsageError("No native token amount or token transfers provided")
+
+        if token_transfers and data:
+            raise BadUsageError("Can't set data field when sending esdt tokens")
+
+        if native_amount and not token_transfers:
+            return self.create_transaction_for_native_token_transfer(
+                sender=sender,
+                receiver=receiver,
+                native_amount=native_amount,
+                data=data.decode() if data else None
+            )
+
+        if token_transfers and not native_amount:
+            return self.create_transaction_for_esdt_token_transfer(
+                sender=sender,
+                receiver=receiver,
+                token_transfers=token_transfers
+            )
+
+        # if the method does not return until here it means both native_amount and token_transfers have been provided
+        # the two lines bellow are only to get rid of pylance warning
+        native_amount = native_amount if native_amount else 0
+        token_transfers = token_transfers if token_transfers else []
+
+        transfers = deepcopy(token_transfers)
+        transfers = list(transfers)
+
+        native_token = Token(EGLD_IDENTIFIER_FOR_MULTI_ESDTNFT_TRANSFER)
+        native_tranfer = TokenTransfer(native_token, native_amount)
+        transfers.append(native_tranfer)
```

Comment:
Maybe we can extract this to a private function such as `_create_native_transfer(): TokenTransfer`.

Additionally, optionally, do the check for native amount before calling `create_transaction_for_esdt_token_transfer` (and only call it once) - the warning should disappear, as well.

Reply:
Very good suggestion, thank you.